src/routes/docs/tutorials/nextjs/step-6/+page.markdoc

```diff
@@ -0,0 +1,195 @@
+---
+layout: tutorial
+title: Create expense list
+description: Fetch user expenses from Appwrite database and display them in a list.
+step: 6
+---
+# Create list API {% #create-list-api %} 
+We will create a new API endpoint that will fetch the user expenses from the database. Create a new file `/src/app/api/expense/list/route.js` and add the following code.
+
+```js
+import { Query } from "appwrite";
+import { headers } from "next/headers";
+import { NextResponse } from "next/server";
+
+import {
+  EXPENSE_COLLECTION_ID,
+  EXPENSE_DATABASE_ID,
+  account,
+  database,
+} from "@/config/appwrite";
+
+export async function POST(req) {
+  const body = await req.json();
+  const { userId } = body;
+  const headersList = headers();
+  const jwt = headersList.get("jwt");
+
+  if (!userId || !jwt) {
+    return new NextResponse("Unauthorized", { status: 401 });
+  }
+  account.client.setJWT(jwt);
+
+  try {
+    const items = await database.listDocuments(
+      EXPENSE_DATABASE_ID,
+      EXPENSE_COLLECTION_ID,
+      [Query.equal("userId", [userId]), Query.orderDesc("$createdAt")]
+    );
+    return new NextResponse(JSON.stringify({ items: items.documents }));
+  } catch (error) {
+    console.log(error);
+    return new NextResponse("Internal server error", {
+      status: 500,
+      error: error.message,
+    });
+  }
+}
+```
+
+With Appwrite's `listDocuments` method we can fetch the documents from the database. It will create a new API endpoint `http://localhost:3000/api/expense/list` that will return the user expenses.
+
+We are also using Appwrite's `Query` class to filter the documents. We have used the `Query.equal` method to filter the documents by `userId`. We have also used the `Query.orderDesc` method to order the documents by `$createdAt` in descending order.
+
+
+# Expense list component {% #expense-list-component %}
+
+Let's fetch the user expenses from the database and display them in a list. Open the `app/page.js` file and add the following code.
+
+```js
+"use client";
```

Comment:
For this change, the problem I still have is how confusing this example is.

The only part hidden is old code, but where did the rest of the code change? Can you break up this example?

Is there another way to express this change?

Maybe just show what's changed and use placeholder functions/empty functions and commented HTML placeholders in previous steps to make it more clear?